### PR TITLE
test: Remove fakeredis & replace `freezegun` with `time-machine`

### LINKED ISSUE #2021: Tests: Remove fakeredis
We still depend on fakeredis, which causes troubles in some cases. It can simply be replaced by the `redis_service` fixture we have in place, which provides a docker based redis service.

## Funding
* If you would like to see an issue prioritized, make a pledge towards it!
* We receive the pledge once the issue is completed & verified

<a href="https://polar.sh/litestar-org/litestar/issues/2021">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/litestar/issues/2021/pledge.svg?darkmode=1">
  Fund with Polar
</picture>
</a>

### PR DESCRIPTION
**Pull request by Alc-Alc**
### Pull Request Checklist

[//]: # "Please review the [Litestar contribution guidelines](https://github.com/litestar-org/litestar/blob/main/CONTRIBUTING.rst) for this repository."

- [ ] New code has 100% test coverage
- [ ] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [ ] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR

[//]: # "By submitting this issue, you agree to:"
[//]: # "- follow Litestar's [Code of Conduct](https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md)"
[//]: # "- follow Litestar's [Contribution Guidelines](https://litestar.dev/community/contribution-guide)"

### Description

[//]: # "Please describe your pull request for new release changelog purposes"
[//]: # "Example: 'This pr adds the capability to use server-sent events, and documents it in the usage docs'"
Removes usage of fakeredis and replaces it with a docker based Redis instance.

### Close Issue(s)

[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"
Closes #2021

### CODE REVIEW

**Comment by Alc-Alc** [comment]
I took a shot at this, however I am not sure how I could modify this test such that it actually tests it, right now appears to be a false positive as it doesn't actually test the expiration behavior.

https://github.com/litestar-org/litestar/blob/b3dbb2c6373da76fce4f533bc4bdcef86f2f5341/tests/unit/test_stores.py#L103-L111

**Comment by Goldziher** [comment]
Well, we will need to switch to using our docker redis for these tests - you can see an example of how this is done in the channels tests.

**Comment by provinzkraut** [comment]
> Well, we will need to switch to using our docker redis for these tests

It seems to me that this is exactly what this PR does :)

@Alc-Alc I think in those cases all we can (reasonably) do is set a low expiry time and mark the test as flaky.

File: tests/unit/test_stores.py [hunk 1]
```diff
@@ -166,7 +169,7 @@ async def test_expires_in(store: Store, frozen_datetime: FrozenDateTimeFactory)
     assert expiration is not None
     assert math.ceil(expiration / 10) == 1
 
-    frozen_datetime.tick(12)
+    await store._redis.expire(f"{store.namespace}:foo", 0)
```
Comment (provinzkraut) [inline tests/unit/test_stores.py:172]: This needs a check to see if it's actually a redis store

Comment (Alc-Alc) [reply tests/unit/test_stores.py:150]: I did have a check there, but the condition for xfail here https://github.com/litestar-org/litestar/pull/2030/files#diff-8d1aade4d9d5b80560a1e0d88ee2bf52ad65ec295584ac7a95c80694536e278eR138 made mypy complain about unreachable code

Comment (provinzkraut) [reply tests/unit/test_stores.py:150]: Oh, I missed that!

**Review by provinzkraut — APPROVED** [review]

**Comment by JacobCoffee** [comment]
@all-contributors add @Alc-Alc for tests

**Pull request merged**